libs/core/src/lib/resizable-card-layout/resizable-card-layout.component.ts

```diff
@@ -0,0 +1,497 @@
+        }
+
+        if (columnsStart === columnsSpan) {
+            this._columnsHeight[columnsStart] = columnHeight;
+        }
+    }
+
+    /**
+     * @hidden Try to set card at available heights
+     */
+    private _setCardPositionValues(card: ResizableCardItemComponent, index: number): void {
+        if (index === 0) {
+            card.left = 0 + this._paddingLeft;
+            card.top = 0;
+            card.startingColumnPosition = 0;
+            return;
+        }
+
+        const uniqueHeights = this._getSortedUniqueHeights();
+        let cardPositioned = false;
+        for (const height of uniqueHeights) {
+            if (!cardPositioned) {
+                cardPositioned = this._isPositionSetSuccess(height, card);
+            } else {
+                break;
+            }
+        }
+    }
+
+    /**
+     * @hidden try to set card at given height
+     */
+    private _isPositionSetSuccess(height: number, card: ResizableCardItemComponent): boolean {
+        const columnPositions = new Array();
+        let index = 0;
+        for (const columnHeight of this._columnsHeight) {
+            index++;
+            if (columnHeight === height) {
+                columnPositions.push(index);
+            }
+        }
+
+        // check for each card position, starting from leftmost
+        let isFitting = false;
+        let startingColumnPosition = -1;
+
+        // try to fit as left as possible from the column position
+        startingColumnPosition = this._fitLeftFromColumnPosition(card, columnPositions);
+        // if not moving towards left. start from column position and check if fits.
+        // try to set towards right from available card position
+        if (startingColumnPosition === -1) {
+            startingColumnPosition = this._fitRight(card, columnPositions, height);
+        }
+
+        if (startingColumnPosition !== -1) {
+            isFitting = true;
+            card.left =
+                startingColumnPosition * HorizontalResizeStep +
+                this._paddingLeft +
+                (startingColumnPosition > 0 ? verticalResizeStep * startingColumnPosition : 0);
+            card.top = height + (height > 0 ? verticalResizeStep : 0);
+            card.startingColumnPosition = startingColumnPosition;
+        }
+        return isFitting;
+    }
+
+    /** @hidden Try to start card position from left most fit position */
+    private _fitLeftFromColumnPosition(card: ResizableCardItemComponent, columnPositions: Array<number>): number {
+        // check for each card position, starting from leftmost
+        let isFitting = false;
+        let startingColumnPosition = -1;
+
+        // start from previous indexes
+        const cardColSpan = Math.floor(card.cardWidth / HorizontalResizeStep);
+
+        // try to set towards left from available card position
+        // eg. [1, 2, 3, 4] columnsPositions
+        for (const columnPosition of columnPositions) {
+            if (!isFitting) {
```

Comment:
```suggestion
        for (const columnPosition of columnPositions) {
            if (isFitting) {
                break;
            }
```

Can we do like that to decrease nesting?

Reply:
fixed